src/sections/create-dataset/CreateDatasetFactory.tsx

```diff
@@ -1,11 +1,27 @@
-import { CreateDatasetForm } from './CreateDatasetForm'
 import { ReactElement } from 'react'
+import { useSearchParams } from 'react-router-dom'
+import { CreateDatasetForm } from './CreateDatasetForm'
 import { DatasetJSDataverseRepository } from '../../dataset/infrastructure/repositories/DatasetJSDataverseRepository'
+import { MetadataBlockInfoJSDataverseRepository } from '../../metadata-block-info/infrastructure/repositories/MetadataBlockInfoJSDataverseRepository'
 
 const repository = new DatasetJSDataverseRepository()
+const metadataBlockInfoRepository = new MetadataBlockInfoJSDataverseRepository()
 
 export class CreateDatasetFactory {
   static create(): ReactElement {
-    return <CreateDatasetForm repository={repository} />
+    return <CreateDatasetFormWithSearchParams />
   }
 }
+
+function CreateDatasetFormWithSearchParams() {
+  const [searchParams] = useSearchParams()
+  const collectionId = searchParams.get('collectionId') ?? undefined
+
+  return (
+    <CreateDatasetForm
+      repository={repository}
+      metadataBlockInfoRepository={metadataBlockInfoRepository}
+      collectionId={collectionId}
+    />
+  )
+}
```

Comment:
I understand, the reason I needed to read the collection id from the url was for the case where the user clicks on the "New Dataset" button from a different collection than the root. I think it will also be useful to pre-populate the Host Dataverse entry.

Reply:
I see your point, makes sense, thanks!